src/llama.cpp

```diff
@@ -17379,6 +17402,11 @@ struct llama_context * llama_new_context_with_model(
         params.flash_attn = false;
     }
 
+    if (params.flash_attn && model->arch == LLM_ARCH_GEMMA2) {
+        LLAMA_LOG_WARN("%s: flash_attn is not compatible with Gemma2 - forcing off\n", __func__);
+        params.flash_attn = false;
+    }
```

Comment:
This should check for `attn_soft_cap` instead of the arch. Eg:
```suggestion
    if (params.flash_attn && model->hparams.attn_soft_cap) {
        LLAMA_LOG_WARN("%s: flash_attn is not compatible with attn_soft_cap - forcing off\n", __func__);
        params.flash_attn = false;
    }

```